### ISSUE TITLE
add_rows does not respect stacked panels 

### ISSUE THREAD

**Issue by lrose1** [vincentarelbundock/modelsummary#626]
Say I want to add a row before Panel B:

```
library(modelsummary)
library(tibble)

rows <- tribble(~term,          ~OLS,  ~Logit,
                'Info',         '???', 'XYZ')
attr(rows, 'position') <- c(6)

gm <- c("r.squared", "nobs", "rmse")

panels <- list(
  list(
    lm(mpg ~ 1, data = mtcars),
    lm(mpg ~ qsec, data = mtcars)
  ),
  list(
    lm(hp ~ 1, data = mtcars),
    lm(hp ~ qsec, data = mtcars)
  )
)

modelsummary(
  panels,
  shape = "rbind",
  gof_map = gm, add_rows = rows)
```
The RMSE is pushed below to the next panel:

<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
 <thead>
  <tr>
   <th style="text-align:left;">   </th>
   <th style="text-align:center;">  (1) </th>
   <th style="text-align:center;">  (2) </th>
  </tr>
 </thead>
<tbody>
  <tr grouplength="6"><td colspan="3" style="border-bottom: 1px solid;"><em>Panel A</em></td></tr>
<tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1"> (Intercept) </td>
   <td style="text-align:center;"> 20.091 </td>
   <td style="text-align:center;"> -5.114 </td>
  </tr>
  <tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1">  </td>
   <td style="text-align:center;"> (1.065) </td>
   <td style="text-align:center;"> (10.030) </td>
  </tr>
  <tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1"> qsec </td>
   <td style="text-align:center;">  </td>
   <td style="text-align:center;"> 1.412 </td>
  </tr>
  <tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1">  </td>
   <td style="text-align:center;">  </td>
   <td style="text-align:center;"> (0.559) </td>
  </tr>
  <tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1"> R2 </td>
   <td style="text-align:center;"> 0.000 </td>
   <td style="text-align:center;"> 0.175 </td>
  </tr>
  <tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1"> Info </td>
   <td style="text-align:center;"> ??? </td>
   <td style="text-align:center;"> XYZ </td>
  </tr>
  <tr grouplength="6"><td colspan="3" style="border-bottom: 1px solid;"><em>Panel B</em></td></tr>
<tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1"> RMSE </td>
   <td style="text-align:center;"> 5.93 </td>
   <td style="text-align:center;"> 5.39 </td>
  </tr>
  <tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1"> (Intercept) </td>
   <td style="text-align:center;"> 146.688 </td>
   <td style="text-align:center;"> 631.704 </td>
  </tr>
  <tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1">  </td>
   <td style="text-align:center;"> (12.120) </td>
   <td style="text-align:center;"> (88.700) </td>
  </tr>
  <tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1"> qsec </td>
   <td style="text-align:center;">  </td>
   <td style="text-align:center;"> -27.174 </td>
  </tr>
  <tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1">  </td>
   <td style="text-align:center;">  </td>
   <td style="text-align:center;"> (4.946) </td>
  </tr>
  <tr>
   <td style="text-align:left;padding-left: 2em;box-shadow: 0px 1px" indentlevel="1"> R2 </td>
   <td style="text-align:center;box-shadow: 0px 1px"> 0.000 </td>
   <td style="text-align:center;box-shadow: 0px 1px"> 0.502 </td>
  </tr>
  <tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1"> RMSE </td>
   <td style="text-align:center;"> 67.48 </td>
   <td style="text-align:center;"> 47.64 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> Num.Obs. </td>
   <td style="text-align:center;"> 32 </td>
   <td style="text-align:center;"> 32 </td>
  </tr>
</tbody>
</table>

**Comment by lrose1** [comment]
I will add there seems to be something in the documentation [here](https://vincentarelbundock.github.io/modelsummary/articles/modelsummary.html#shape-pivot-groups-panels-and-stacks), but the added function does not seem to called.

**Comment by vincentarelbundock** [comment]
Thanks a lot for the report and the very useful reproducible example!

This is just to confirm that I can reproduce the bug on my computer.

I will fix this as soon as I find a moment (but life is busy these days).

**Comment by vincentarelbundock** [comment]
Thanks again for the report @lrose1 . This problem should be fixed in version 1.4.0.9001, which you can now install from Github (but not CRAN yet). Make sure you restart R after install for the change to take effect.

**Comment by lrose1** [comment]
Confirmed this works, thank you for this and for all your awesome work!
